### PR TITLE
Multi-root workspace support and Fix TextMate grammar bugs

### PR DESCRIPTION
**Pull request by mattmasson**
### Grammar Fixes

Validated `syntaxes/powerquery.tmLanguage.json` against the [official M language specification](https://learn.microsoft.com/en-us/powerquery-m/m-spec-lexical-structure) and fixed 4 bugs:

#### 1. Dotted identifiers not matched beyond 2 characters after the dot
The `Identifier` regex was missing a `*` quantifier on the continuation character class in the dotted-identifier part. This caused `Table.AddColumn` to be split into `"Table"` (scoped) + `".AddColumn"` (unscoped) instead of being recognized as a single identifier. Per the M spec, dotted identifiers like `Table.AddColumn` are a single `regular-identifier`.

**Before:** `\.[start-char][continuation-char]` — only matched 2 chars after dot (e.g. `.AB`)  
**After:** `\.[start-char][continuation-char]*` — matches any length (e.g. `.AddColumn`)

#### 2. `<=` and `>=` operators split into two tokens
The operator regex alternation was `(<>|<|>|<=|>=)` — since `<` appears before `<=`, the regex engine matched `<` first, leaving `=` as a separate assignment token. Same for `>=`.

**Before:** `(<>|<|>|<=|>=)` → `<=` tokenized as `<` + `=`  
**After:** `(<>|<=|>=|<|>)` — longest-match-first ordering

#### 3. Missing `??` null-coalescing operator
The `??` operator is defined in the M spec as an `operator-or-punctuator` but only single `?` was matched. Added `(\\?\\?)` before `(\\?)` and a new `keyword.operator.nullcoalescing.powerquery` scope.

#### 4. Missing `;` semicolon punctuator
Semicolons are used in section documents (`section Foo; shared x = 1;`) and are defined in the M spec as an `operator-or-punctuator`, but had no grammar rule. Added as `punctuation.terminator.powerquery`.

### Grammar Test Suite

Added a comprehensive test suite under `server/src/test/grammar/` using `vscode-textmate` + `vscode-oniguruma` (the standard approach for testing TextMate grammars outside VS Code) integrated with the existing mocha/chai test infrastructure.

**166 new tests** across 7 test files:

| File | Coverage |
|------|----------|
| `keywords.test.ts` | All M keywords including `#keyword` forms (`#binary`, `#date`, etc.) |
| `identifiers.test.ts` | Regular, dotted (`Table.AddColumn`), quoted (`#"name"`), inclusive (`@`), implicit (`_`), intrinsic (`#shared`) |
| `literals.test.ts` | Strings with escapes, integers, hex, decimals, floats, `#infinity`, `#nan`, `true`/`false`, `null` |
| `operators.test.ts` | All operators (`=`, `<>`, `<=`, `>=`, `=>`, `??`, `..`, `...`) and punctuators (`,`, `;`, `()`, `{}`, `[]`) |
| `comments.test.ts` | Line (`//`) and block (`/* */`) comments, edge cases |
| `types.test.ts` | All 19 primitive types, `optional`/`nullable` modifiers |
| `expressions.test.ts` | Real-world M patterns: `let`/`in`, `if`/`then`/`else`, `each`, `try`/`catch`, functions, records, sections |

### Other Changes

- Added `vscode-textmate` and `vscode-oniguruma` as server devDependencies
- Updated `server/.mocharc.json` to include compiled grammar tests
- Added `skipLibCheck` to `ts-node` config in `server/tsconfig.json` (needed for `vscode-oniguruma` WebAssembly types)

### CODE REVIEW

File: server/tsconfig.json [hunk 1]
```diff
@@ -5,5 +5,10 @@
         "rootDir": "src"
     },
     "include": ["src/**/*.ts"],
-    "exclude": ["node_modules"]
+    "exclude": ["node_modules"],
+    "ts-node": {
+        "compilerOptions": {
+            "skipLibCheck": true
```
Comment (JordanBoltonMN) [inline server/tsconfig.json:12]: Do we want this?

File: server/src/test/standardLibrary.test.ts [hunk 2]
```diff
@@ -253,3 +253,66 @@ describe(`setLibrarySymbols`, () => {
         expect(ExternalLibraryUtils.getSymbols().length).to.equal(0, "expected 0 libraries");
     });
 });
+
+describe(`ExternalLibraryUtils overlap detection`, () => {
+    const symbolA: PQLS.LibrarySymbol.LibrarySymbol[] = JSON.parse(additionalSymbolJsonStr);
+
+    afterEach(() => {
+        ExternalLibraryUtils.removeLibraries(ExternalLibraryUtils.getRegisteredModuleNames() as string[]);
+    });
```
[the remark this answers is not in the document]
Comment (JordanBoltonMN) [reply server/src/test/standardLibrary.test.ts:262]: I think this is correct?

File: .github/copilot-instructions.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# Copilot Instructions
```
Comment (JordanBoltonMN) [inline .github/copilot-instructions.md:1]: Was there a typo, or did the naming convention change (i.e. the missing leading dot)

File: server/tsconfig.json (hunk 1 above)
Comment (mattmasson) [reply server/tsconfig.json:12]: yeah - it's needed for the new unit tests. it looks like ts-node uses a different version of typescript than we are, and the settings are different. there was some copilot suggestions on things we could move to, but that could be done in a follow up.

File: .github/copilot-instructions.md (hunk 3 above)
Comment (mattmasson) [reply .github/copilot-instructions.md:1]: I think previously you'd configure the path(s) to instructions using settings, and then they switched it to be convention based by default. i created the original file - this one was created with the agency /init switch.

**Comment by RedCMD** [comment]
could also use an existing TextMate grammar test tool
https://www.npmjs.com/package/textmate-grammar-test
instead of coding up your own

File: client/src/extension.ts [hunk 4]
```diff
@@ -85,18 +96,37 @@ export function deactivate(): Thenable<void> | undefined {
...
+
+        const additionalSymbolsDirectories: string[] | undefined = config.get(
+            ConfigurationConstant.AdditionalSymbolsDirectories,
+        );
+
+        await librarySymbolManager.refreshSymbolDirectories(additionalSymbolsDirectories ?? []);
+
+        return;
+    }
+
```
[the remark this answers is not in the document]
Comment (mattmasson) [reply client/src/extension.ts:114]: Added removeAllSymbols() before per-folder registration to clear stale global symbols

File: server/src/library/externalLibraryUtils.ts [hunk 5]
```diff
@@ -21,4 +21,34 @@ export function removeLibraries(libraryNames: ReadonlyArray<string>): void {
...
+            symbolToModules.set(symbol.name, modules);
+        }
+    }
+
+    const overlaps: Map<string, ReadonlyArray<string>> = new Map();
+
+    for (const [symbolName, modules] of symbolToModules) {
+        if (modules.length > 1) {
+            overlaps.set(symbolName, modules);
+        }
```
[the remark this answers is not in the document]
Comment (mattmasson) [reply server/src/library/externalLibraryUtils.ts:48]: Changed string[] → Set<string> to prevent false overlap reports when a module has duplicate symbol names

File: server/.mocharc.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
     "require": ["ts-node/register"],
-    "extensions": ["ts"],
-    "spec": "src/test/**/*.test.ts"
+    "extensions": ["ts", "js"],
+    "spec": ["src/test/*.test.ts", "lib/test/grammar/**/*.test.js"]
```
[the remark this answers is not in the document]
Comment (mattmasson) [reply server/.mocharc.json:4]: Running grammar tests from TS source fails because this project uses module: "node16" (ESM) and the grammar tests use .js extension imports per Node16 convention. `ts-node/register` can't resolve .js → .ts in ESM mode. Keeping grammar tests running from compiled `lib/` is correct; the `skipLibCheck` fix ensures `tsc` succeeds before mocha runs them.

File: client/src/librarySymbolManager.ts [hunk 7]
```diff
@@ -162,17 +166,70 @@ export class LibrarySymbolManager {
...
+            const normalized: string = path.normalize(directory);
+
+            if (directory !== normalized) {
+                this.clientTrace?.info(`Normalized symbol file path '${directory}' => '${normalized}'`);
+            }
+
+            return vscode.Uri.file(normalized);
+        });
+
+        const dedupedDirectoryUris: ReadonlyArray<vscode.Uri> = Array.from(new Set(normalizedDirectoryUris));
```
Comment (JordanBoltonMN) [inline client/src/librarySymbolManager.ts]: This is an object so this won't dedupe anything.

**Review by JordanBoltonMN — APPROVED** [review]
